app/src/main/java/com/tari/android/wallet/di/ApplicationModule.kt

```diff
@@ -69,7 +81,28 @@ internal class ApplicationModule(
 
     @Provides
     @Singleton
-    fun provideClipboardManager(context: Context): ClipboardManager {
-        return context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
-    }
-}
\ No newline at end of file
+    fun provideClipboardManager(context: Context): ClipboardManager =
+        context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+
+    @Provides
+    @Singleton
+    fun provideBiometricAuthService(context: Context): BiometricAuthenticationService =
+        BiometricAuthenticationService(
+            ContextCompat.getMainExecutor(context),
+            BiometricManager.from(context),
+            context.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
+        )
+
+    @Provides
+    @Singleton
+    fun provideWalletBackup(wd: File) = WalletBackup.defaultStrategy(wd)
```

Comment:
`wd` is not clear (is it `workingDirectory`), let's avoid abbreviations/acronyms when it's not very much common use.

Reply:
I think you can go for `@Named` for this one to have safer injection.

Reply:
there is no type conflicts currently, so it is possible to introduce complexity when it is necessary